### PR TITLE
fix: getPopupContainer not work inject by ConfigProvider (#40781)

### LINKED ISSUE #40781: Filter submenu is rendered outside the target container when used with ConfigProvider
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/s/filter-search-antd-4-24-8-forked-t5xse4)

### Steps to reproduce

1. Load the sandbox
2. Click the filter button in the "Name" column
3. Hover over the "Submenu"
4. Inspect the first item in the menu

### What is expected?

The submenu is rendered within the target container.

### What is actually happening?

The submenu is rendered outside the target container (attached to the body instead of the target container).

| Environment | Info |
| --- | --- |
| antd | 4.24.4 |
| React | 18 |
| System | MacOS Ventura 13.2 |
| Browser | All |

---
The submenu is rendered correctly within the container for simple dropdowns. Please see https://codesandbox.io/s/cascading-menu-antd-4-24-8-forked-zeyndq.

Ref: UIEN-3286

### PR DESCRIPTION
**Pull request by RedJue**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
fix #40781

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    fix getPopupContainer not work inject by ConfigProvider      |
| 🇨🇳 Chinese |   修复 ConfigProvider 注入的  getPopupContainer属性 不生效的问题       |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

File: components/table/__tests__/Table.test.tsx [hunk 1]
```diff
@@ -369,4 +369,37 @@ describe('Table', () => {
...
+          </div>
+        </ConfigProvider>
+      );
+    };
+
+    const { container } = render(<Demo />);
+
+    fireEvent.click(container.querySelector('.ant-table-filter-trigger')!);
+    await waitFakeTimer();
+    expect(container.firstChild).toMatchSnapshot();
```
Comment (zombieJ) [inline components/table/__tests__/Table.test.tsx:402]: 这个就不要比 snapshot 了，直接测一下弹出的在不在这个容器了

Comment (RedJue) [reply components/table/__tests__/Table.test.tsx:402]: 好

File: components/table/InternalTable.tsx [hunk 2]
```diff
@@ -323,7 +324,7 @@ function InternalTable<RecordType extends object = any>(
     dropdownPrefixCls,
     mergedColumns,
     onFilterChange,
-    getPopupContainer,
+    getPopupContainer: getPopupContainer || getContextPopupContainer,
```
Comment (afc163) [inline components/table/InternalTable.tsx:327]:
```suggestion
    getPopupContainer: getPopupContainer ?? getContextPopupContainer,
```

File: components/table/InternalTable.tsx [hunk 3]
```diff
@@ -408,7 +409,7 @@ function InternalTable<RecordType extends object = any>(
     expandType,
     childrenColumnName,
     locale: tableLocale,
-    getPopupContainer,
+    getPopupContainer: getPopupContainer || getContextPopupContainer,
```
Comment (afc163) [inline components/table/InternalTable.tsx:412]:
```suggestion
    getPopupContainer: getPopupContainer ?? getContextPopupContainer,
```

**Comment by afc163** [comment]
🔗 Related issue link

这个 issue 链接留错了？

File: components/table/InternalTable.tsx (hunk 2 above)
Comment (RedJue) [reply components/table/InternalTable.tsx:327]: 这样写是为了规避什么场景？

Comment (afc163) [reply components/table/InternalTable.tsx:327]: 手动声明 getPopupContainer={null} 不应该回退到 ConfigProvider 的默认值上。

**Comment by RedJue** [comment]
> 🔗 Related issue link
> 
> 
> 
> 这个 issue 链接留错了？

好像是错了，没留意直接选的，我改一下

File: components/table/InternalTable.tsx (hunk 2 above)
Comment (RedJue) [reply components/table/InternalTable.tsx:327]: 懂了，优先级原因，我看仓库里也有其他地方这样写的，也一并修正一下？

Comment (RedJue) [reply components/table/InternalTable.tsx:327]: 这样写好像也会会回退到默认值上去，null没法走??之前的逻辑

**Review by afc163 — APPROVED** [review]

File: components/table/InternalTable.tsx (hunk 2 above)
Comment (zombieJ) [reply components/table/InternalTable.tsx:327]: null 和 undefined 都会被 ?? 命中

**Pull request merged**